account_payment_mode_auto_reconcile/tests/test_partner_auto_reconcile.py

```diff
@@ -0,0 +1,235 @@
+        refund = self.refund.copy()
+        refund.invoice_line_ids.write({'price_unit': 100})
+        refund.action_invoice_open()
+        new_invoice = self.invoice_copy.copy()
+        new_invoice.action_invoice_open()
+        # Only reconcile 1000 refund manually
+        new_invoice_credits = json.loads(
+            new_invoice.outstanding_credits_debits_widget
+        ).get('content')
+        for cred in new_invoice_credits:
+            if cred.get('amount') == 100.0:
+                new_invoice.assign_outstanding_credit(cred.get('id'))
+        self.assertEqual(new_invoice.residual, 1400.0)
+        # Assign payment mode adds the outstanding credit of 1000.0
+        new_invoice.write({'payment_mode_id': self.payment_mode.id})
+        self.assertEqual(new_invoice.residual, 400.0)
+        # Remove payment mode only removes automatically added credit
+        new_invoice.write({'payment_mode_id': False})
+        self.assertEqual(new_invoice.residual, 1400.0)
+
+        # use the same payment partially on different invoices.
+        other_invoice = self.invoice.copy()
+        other_invoice.invoice_line_ids.write({
+            'price_unit': 200,
+        })
+        other_invoice.write({
+            'payment_mode_id': self.payment_mode.id,
+        })
+        other_invoice.action_invoice_open()
+        self.assertEqual(other_invoice.state, 'paid')
+        # since 200 were assigned on other invoice adding auto-rec payment mode
+        # on new_invoice will reconcile 800 and residual will be 600
+        new_invoice.write({'payment_mode_id': self.payment_mode.id})
+        self.assertEqual(new_invoice.residual, 600.0)
+        # Removing the payment mode should not remove the partial payment on
+        #  the other invoice
+        new_invoice.write({'payment_mode_id': False})
+        self.assertEqual(new_invoice.residual, 1400.0)
+        self.assertEqual(other_invoice.state, 'paid')
+
+    def test_invoice_auto_reconcile_same_journal(self):
+        """Check reconciling credits on same journal."""
+        self.payment_mode.auto_reconcile_same_journal = True
+        auto_rec_invoice = self.invoice.copy({
+            'payment_mode_id': self.payment_mode.id,
+        })
+        auto_rec_invoice.write({
+            'invoice_line_ids': [(0, 0, {
+                'product_id': self.product.id,
+                'name': self.product.name,
+                'price_unit': 500.0,
+                'quantity': 1,
+                'account_id': self.acc_rev.id,
+            })]
+        })
+        self.assertTrue(self.payment_mode.auto_reconcile_outstanding_credits)
+        self.assertEqual(self.invoice_copy.residual, 1500)
+        auto_rec_invoice.action_invoice_open()
+        self.assertEqual(auto_rec_invoice.residual, 500)
+
+    def test_invoice_auto_reconcile_different_journal(self):
+        """Check not reconciling credits on different journal."""
+        self.payment_mode.auto_reconcile_same_journal = True
```

Comment:
```suggestion
        self.payment_mode.auto_reconcile_same_journal = False
```
?

Reply:
No this is correct the test is checking that when the option is set credits on a different journal is not reconciled.